src/main/java/pro/belbix/ethparser/web3/harvest/HardWorkCalculator.java

```diff
@@ -0,0 +1,132 @@
+package pro.belbix.ethparser.web3.harvest;
+
+import org.springframework.stereotype.Service;
+
+import pro.belbix.ethparser.dto.v0.HardWorkDTO;
+import pro.belbix.ethparser.dto.v0.HarvestDTO;
+import pro.belbix.ethparser.repositories.v0.HardWorkRepository;
+import pro.belbix.ethparser.repositories.v0.HarvestRepository;
+import pro.belbix.ethparser.web3.EthBlockService;
+
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.HashMap;
+import java.util.List;
+import java.util.stream.Collectors;
+
+@Service
+public class HardWorkCalculator {
+
+    private final double ETH_ESTIMATE = 0.1;
+    private final HarvestRepository harvestRepository;
+    private final HardWorkRepository hardworkRepository;
+    private final EthBlockService ethBlockService;
+
+    public HardWorkCalculator(HarvestRepository harvestRepository, HardWorkRepository hardWorkRepository, EthBlockService ethBlockService) {
+        this.harvestRepository = harvestRepository;
+        this.hardworkRepository = hardWorkRepository;
+        this.ethBlockService = ethBlockService;
+    }
+
+    public double calculateTotalHardWorksFeeByOwner(String ownerAddress) {
+        long lastBlock = ethBlockService.getLastBlock();
+        long lastBlockDate = ethBlockService.getTimestampSecForBlock(null, lastBlock);
+        
+        HashMap<String, ArrayList<long[]>> blockRangeByVault = new HashMap<>();
+        List<HarvestDTO> harvests = harvestRepository.fetchAllByOwner(ownerAddress, 0, lastBlockDate);
+        
+        harvests
+            .stream()
+            .filter(harvest -> !harvest.getVault().startsWith("PS"))
```

Comment:
ah, I forgot that PS was excluded from this table a long time ago
but nor we have iPS, better exclude it
and please, add an exclude condition in HardWorkDbService.fillExtraInfo for iPS

Reply:
> ah, I forgot that PS was excluded from this table a long time ago

I still see there are recent records with this vault in the table. So I guess excluding the PS is still needed.

One question though, why should the PS be excluded? Are they better to be deleted from the table to avoid the overheads in the related newly added functions in the future?